File: app/src/main/java/com/manimarank/spell4wiki/models/WikiBaseData.java
package com.manimarank.spell4wiki.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class WikiBaseData {
    @SerializedName("category_common")
    @Expose
    private List<String> categoryCommon = null;
    @SerializedName("language_wise_data")
    @Expose
    private List<WikiLanguage> languageWiseData = null;

    @SerializedName("update_content")
    @Expose
    private UpdateApp updateApp = null;

    @SerializedName("fetch_config")
    @Expose
    private FetchConfig fetchConfig = null;

    public List<String> getCategoryCommon() {
        return categoryCommon;
    }

    public void setCategoryCommon(List<String> categoryCommon) {
        this.categoryCommon = categoryCommon;
    }

    public List<WikiLanguage> getLanguageWiseData() {
        return languageWiseData;
    }

    public void setLanguageWiseData(List<WikiLanguage> languageWiseData) {
        this.languageWiseData = languageWiseData;
    }

    public UpdateApp getUpdateApp() {
        return updateApp;
    }

    public FetchConfig getFetchConfig() {
        return fetchConfig;
    }
}

File: app/src/main/java/com/manimarank/spell4wiki/data/model/WikiBaseData.kt
package com.manimarank.spell4wiki.data.model

import com.google.gson.annotations.SerializedName

/**
 * Model Class for Wiki Base Data
 * @property categoryCommon List<String>?
 * @property languageWiseData List<WikiLanguage>?
 * @property updateApp UpdateApp?
 * @property fetchConfig FetchConfig?
 * @constructor
 */
data class WikiBaseData(
    @SerializedName("category_common")
    var categoryCommon: List<String>? = null,
    @SerializedName("language_wise_data")
    var languageWiseData: List<WikiLanguage>? = null,
    @SerializedName("update_content")
    val updateApp: UpdateApp? = null,
    @SerializedName("fetch_config")
    val fetchConfig: FetchConfig? = null
)
